### PR TITLE
#162163021   A user is able to see the number of times his article has been viewed

### PR DESCRIPTION
**Pull request by kizzanaome**
### What does this PR do?
- Allows a user who is an author in this case to see the number of views on a single article

### Description of Task to be completed?
- Given am an author, I should be able to view the number of times my article has been viewed.
Whenever a reader sends a get request to my article endpoint, the number of views increase.
If i the author sends a get request to  my own article, the number of views dont't increase.
### How should this be manually tested?
First clone this repository
```
$ git clone https://github.com/andela/ah-backend-poseidon
$ cd ah-backend-poseidon
```
Create virtual environment and install it
```
$ python3 -m venv venv
$ source/venv/bin/activate
```
Then install all the necessary dependencies
```
pip install -r requirements.txt
```
Run the application
At the terminal or console type
```
 $ python manage.py make migrations
 $ python manage.py migrate
 $ python manage.py run server
```
Head over to postman and first post an article through this endpoint /api/articles/
Use the fomart bellow to post an article.
```
{
  "article": {
    "title": "Who is he?",
    "description": "He has a bald head",
    "body": "It is believed he will rule Uganda to heaven",
     "tags": ["python", "programming", "technology"]
  }
}
```

Bellow is a fomart you will see in postman after posting an article
```{
    "articles": {
        "slug": "who-is-he-he-has-20181212152330656837",
        "title": "Who is he?",
        "description": "He has a bald head",
        "body": "It is believed he will rule Uganda to heaven",
        "created_on": "2018-12-12T15:23:30.657217Z",
        "average_rating": 0,
        "updated_on": "2018-12-12T15:23:30.657248Z",
        "image_url": null,
         "view_count": 0,
        "author": {
            "username": "naomecoG",
            "email": "naomekizzan@gmail.com",
            "bio": "",
            "image": null,
            "following": false
        },
        "tags": [
            "programming",
            "technology",
            "python"
        
    }
}
```

The returned article comes with a view_counts field which keeps incrementing its self when ever another user views this article.

### What are the relevant pivotal tracker stories?
#162163021

### CODE REVIEW

File: authors/settings.py [hunk 1]
```diff
@@ -95,7 +96,7 @@
         'ENGINE': 'django.db.backends.postgresql_psycopg2',
         'NAME': os.environ.get('DB_NAME', 'ahdb'),
         'USER': os.environ.get('DB_USER', 'postgres'),
-        'PASSWORD': os.environ.get('PASSWORD', ''),
+        'PASSWORD': os.environ.get('PASSWORD', '1460'),
```
Comment (arthurarty) [inline authors/settings.py]: Hey the password should be blank.

**Review by arthurarty — CHANGES_REQUESTED** [review]
Please remove the password from the settings file.

**Comment by katunold** [comment]
**Your version**: `Create virtual environment and install it`
**My suggestion**: `Create a Virtual environment and activate it`

**Comment by katunold** [comment]
**Your version**: $ python manage.py `run server`
**My suggestion**: $ python manage.py `runserver`

**Comment by katunold** [comment]
**Your version**: $ python manage.py `make migrations`
**My suggestion**: $ python manage.py `makemigrations`

**Comment by katunold** [comment]
**Your version**: its self `when ever` another user views this article.
**My suggestion**: its self `whenever` another user views this article.

File: authors/apps/profiles/models.py [hunk 2]
```diff
@@ -20,6 +20,8 @@ class Profile(models.Model):
     favourites = models.ManyToManyField(
         'article.Article',
         related_name='favourite_by')
+    # view_reads = models.IntegerField()
```
Comment (Luleherll) [inline authors/apps/profiles/models.py]: Please remove commented out code.

Comment (Luleherll) [inline authors/apps/profiles/models.py]: Please remove commented out code.

File: authors/apps/article/models.py [hunk 3]
```diff
@@ -35,6 +35,8 @@ class to define the model of an article
 
     user_rating = models.CharField(max_length=10, default='0')
 
+    
+
```
Comment (katunold) [inline authors/apps/article/models.py]: I don't think you need these 2 empty lines

File: authors/apps/tests/test_articles.py [hunk 4]
```diff
@@ -54,6 +54,7 @@ def test_get_specific_article(self):
         self.assertEqual(response.status_code, 200)
         self.assertIn('articles', response.json())
         self.assertIsInstance(response.json().get("articles"), dict)
+
```
Comment (katunold) [inline authors/apps/tests/test_articles.py]: Remove this extra empty line

**Comment by katunold** [comment]
Please don't forget to squash the commits otherwise, I appreciate the work done 👍

**Review by arthurarty — APPROVED** [review]

File: authors/apps/tests/test_articles.py (hunk 4 above)
Comment (kizzanaome) [reply authors/apps/tests/test_articles.py]: @katunold  i have worked on the feedback

**Comment by malep2007** [comment]
Please fix the profiles apps issue

**Pull request merged**